Shadowlands/TazaveshTheVeiledMarket/Hylbrande.lua

```diff
@@ -54,6 +58,14 @@ function mod:UNIT_SPELLCAST_SUCCEEDED(_, _, _, spellId)
 	end
 end
 
+function mod:PurifyingBurst(args)
+	self:Message(args.spellId, "yellow")
+	self:PlaySound(args.spellId, "info")
+	if self:BarTimeLeft(346766) > 21.8 then -- Sanitizing Cycle
```

Comment:
We recommend against using BarTimeLeft unless it's measuring and changing the duration of the same spell. If the user happens to have this bar disabled, it will affect the timer as it will return 0. It's better to just store the time of cast (`args.time` or `GetTime()`) and do the math on that.

Reply:
That makes sense. Most of the other timers in this module use this BarTimeLeft so I'll look into refactoring it all.